File: leetcode/py/p760-find-anagram-mappings.py
# LeetCode 760. Find Anagram Mappings
from typing import List


class Solution:
    def anagramMappings(self, nums1: List[int], nums2: List[int]) -> List[int]:
        d = {}
        for i, n in enumerate(nums2):
            d[n] = i
        l = [0] * len(nums1)
        for x in range(len(nums1)):
            l[x] = d[nums1[x]]
        return l


if __name__ == "__main__":

    s = Solution()
    tests = (
        ([], [], []),
        ([12, 28, 46, 32, 50], [50, 12, 32, 46, 28], [1, 4, 3, 2, 0]),
        ([84, 46], [84, 46], [0, 1]),
    )

    for test in tests:

        nums1 = test[0]
        nums2 = test[1]
        exp = test[2]
        ans = s.anagramMappings(nums1, nums2)
        if exp != ans:
            print(f"FAIL. nums1: {nums1}, nums2: {nums2}, exp: {exp}, ans: {ans}")

File: leetcode/cpp/p760-find-anagram-mappings.cpp
// LeetCode 1762. Buildings With an Ocean View.
#include <algorithm>
#include <iostream>
#include <vector>

class Solution {
public:
    std::vector<int> anagramMappings(const std::vector<int>& nums1, const std::vector<int>& nums2)
    {
        std::vector<int> v = nums1;
        std::transform(v.begin(), v.end(), v.begin(), [&](const auto& n) {
            return static_cast<int>(std::find(nums2.begin(), nums2.end(), n) - nums2.begin());
        });
        return v;
    }
};

int main()
{
    struct TestCase {
        std::vector<int> nums1;
        std::vector<int> nums2;
        std::vector<int> exp;
    };
    const TestCase testCases[] = {
        { {}, {}, {} },
        { { 12, 28, 46, 32, 50 }, { 50, 12, 32, 46, 28 }, { 1, 4, 3, 2, 0 } },
        { { 84, 46 }, { 84, 46 }, { 0, 1 } },
    };

    Solution s;
    for (const auto& tc : testCases) {
        const auto ans = s.anagramMappings(tc.nums1, tc.nums2);
        if (tc.exp != ans) {
            std::cout << "FAIL. "
                      << "(nums1.size()): " << tc.nums1.size() << ")"
                      << ", exp size: " << tc.exp.size()
                      << ", ans size: " << ans.size() << "\n";
        }
    }
    return 0;
}
